### PR TITLE
Adds support for debugging the P0 CLI

### PR DESCRIPTION
**Pull request by fabgo**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -62,6 +62,7 @@
     "eslint-plugin-promise": "^6.1.1",
     "jest": "^29.7.0",
     "prettier": "^3.2.4",
+    "ts-node": "^10.9.2",
```
Comment (nbrahms) [inline package.json:65]: I don't understand why we use this diff. Shouldn't `ts-node` be globally installed?

**Review by nbrahms — COMMENTED** [review]
🤔 We definitely should have some discussion around changing the TS version.

I also think we should avoid altering the script file itself.

File: package.json (hunk 1 above)
Comment (fabgo) [reply package.json:65]: It's there in the app repo also. So ts-node should be installed as part of dev setup?

File: p0 [hunk 2]
```diff
@@ -13,4 +13,8 @@ process.emit = function (name, data, ...args) {
   return originalEmit.apply(process, arguments);
 };
 
-require(`${__dirname}/dist/index.js`).main();
+if (process.env.NODE_ENV === "dev") {
```
Comment (nbrahms) [inline p0:16]: See the previous comment: https://github.com/p0-security/p0cli/pull/102#discussion_r1757690405

Specifically, running `NODE_ENV=dev p0` should work. I think this breaks that.

File: package.json (hunk 1 above)
Comment (nbrahms) [reply package.json:65]: Disregard this comment; I think you do need `ts-node` in the dev dependencies.

File: .vscode/launch.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "type": "node",
+            "request": "launch",
+            "name": "Launch P0 CLI",
+            "skipFiles": [
+                "<node_internals>/**"
+            ],
+            "program": "${workspaceFolder}/p0",
```
Comment (nbrahms) [inline .vscode/launch.json:14]:
```suggestion
            "program": "-e 'require(\"${workspaceFolder}/src/index.ts\").main()'",
```

File: .vscode/launch.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "type": "node",
+            "request": "launch",
+            "name": "Launch P0 CLI",
```
Comment (nbrahms) [inline .vscode/launch.json:10]: This name is misleading, as this actually uses the P0 CLI to SSH to a specific VM

File: p0 (hunk 2 above)
Comment (fabgo) [reply p0:16]: I can change it to something else. NODE_DBEUG = true for example.

File: .vscode/launch.json (hunk 4 above)
Comment (fabgo) [reply .vscode/launch.json:10]: I removed that.

File: .vscode/launch.json (hunk 3 above)
Comment (fabgo) [reply .vscode/launch.json:14]: Done

File: .vscode/launch.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+            "skipFiles": [
+                "<node_internals>/**"
+            ],
+            "program": "${workspaceFolder}/p0",
```
[the remark this answers is not in the document]
Comment (fabgo) [reply .vscode/launch.json]: Done

File: .vscode/launch.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+            "preLaunchTask": "tsc: build - tsconfig.json",
+            "args": ["ssh","dev-instance-fabian-joya"],
```
[the remark this answers is not in the document]
Comment (fabgo) [reply .vscode/launch.json]: Done

File: p0 [hunk 7]
```diff
@@ -13,4 +13,8 @@ process.emit = function (name, data, ...args) {
   return originalEmit.apply(process, arguments);
 };
 
-require(`${__dirname}/dist/index.js`).main();
+if (process.env.NODE_ENV === 'dev') {
```
[the remark this answers is not in the document]
Comment (fabgo) [reply p0]: Done

File: package.json [hunk 8]
```diff
@@ -35,7 +35,7 @@
     "pluralize": "^8.0.0",
     "semver": "^7.6.0",
     "tmp-promise": "^3.0.3",
-    "typescript": "^4.8.4",
+    "typescript": "^5.6.2",
```
[the remark this answers is not in the document]
Comment (fabgo) [reply package.json]: Reverted

File: p0 (hunk 2 above)
Comment (fabgo) [reply p0]: Reverted

**Review by nbrahms — APPROVED** [review]
lgtm

**Pull request merged**